### PR TITLE
(492) Get docker-compose working locally with new DB

### PR DESCRIPTION
**Pull request by Gweaton**
## Context

https://trello.com/c/VDp60gWQ/492-fix-ui-version-of-the-api-now-that-a-database-has-a-been-added

`script/server` started failing once we added the database to the API, as we weren't starting a postgres database.

## Changes in this PR

This starts all the necessary backing services for running the API and our dependencies locally now that the API has a `seed` environment to start up with data.

## Post-merge checklist

- [ ] This adds or extends a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notifed the API developer(s) of changes to the contract tests (Pact), or the API is already compliant

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -62,7 +86,9 @@ services:
     build: .
     networks:
       - hmpps
-    depends_on: [redis]
+    depends_on:
+      - redis
+      - hmpps-accredited-programmes-api
```
Comment (yndajas) [inline docker-compose.yml:91]: Why does this rely on `hmpps-accredited-programmes-api`?

**Review by yndajas — APPROVED** [review]
Working great locally for me - just one query

File: docker-compose.yml (hunk 1 above)
Comment (Gweaton) [reply docker-compose.yml]: This was used to start the app with docker compose, so I wanted it to wait for the API to start so we could fetch courses, but we never actually use docker compose to start the app. I've removed this whole section now.

File: script/utils/start-backing-services [hunk 2]
```diff
@@ -10,12 +10,12 @@ do
...
-  docker-compose up --scale=app=0 --scale=hmpps-accredited-programmes-api=0 -d
+  docker-compose up --scale=hmpps-accredited-programmes-api=0 -d
 
   echo "==> Stubbing API endpoints..."
 
   npm run api-stubs:reset > /dev/null
   npm run api-stubs:create > /dev/null
 else
-  docker-compose up --scale=app=0 --scale=wiremock=0 -d
+  docker-compose up --scale=wiremock=0 -d
```
Comment (yndajas) [inline script/utils/start-backing-services]: Not sure I fully understand the commit message here, but basically we were always setting `--scale=app=0` when running `script/server` so it was never being used?

**Review by yndajas — APPROVED** [review]
Can we scale postgres down when mocking the API?

Otherwise, looks good

File: script/utils/start-backing-services (hunk 2 above)
Comment (Gweaton) [reply script/utils/start-backing-services]: Yeah, exactly. I've reworded the commit message to hopefully be a bit clearer on this, hope it makes more sense.

**Comment by Gweaton** [comment]
> Can we scale postgres down when mocking the API?

Good idea, I've pushed a new commit for this.

**Pull request merged**